### PR TITLE
refactor(nvim-tree): update settings structure

### LINKED ISSUE #2302: PackerSync/Update error on fresh install
### Problem description

After installing LunarVim and all its dependencies, I get a "Failed to update wbthomason/packer.nvim" error with the following error:

`fatal: reference is not a tree: c5e98e3ca84843dbae47cd8f3a76bc38c6404241`

Can anyone help me Sync/Update Packer properly? I'm unable to use it.

### LunarVim version

c8157a0

### Neovim version (>= 0.6.1)

0.6.1

### Operating system/version

ArchLinux

### Relevant log output

```shell
:
```

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by abzcoding**
# Description

Possibly Fixes #2302

### CODE REVIEW

File: lua/lvim/core/nvimtree.lua [hunk 1]
```diff
@@ -67,6 +73,14 @@ function M.config()
         cmd = "trash",
         require_confirm = true,
       },
+      actions = {
+        change_dir = {
+          global = false,
+        },
+        open_file = {
+          quit_on_open = false,
+        },
```
Comment (kylo252) [inline lua/lvim/core/nvimtree.lua:82]:
```suggestion
        },
      	window_picker = {
	    enable = false,
	    chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890",
	    exclude = {} -- not sure if we need to define this or not
	 },
```
This has replaced `disable_window_picker = 0`. You can keep it enabled if you'd like.

File: lua/lvim/core/nvimtree.lua [hunk 2]
```diff
@@ -67,6 +73,14 @@ function M.config()
         cmd = "trash",
         require_confirm = true,
       },
+      actions = {
+        change_dir = {
+          global = false,
+        },
+        open_file = {
+          quit_on_open = false,
```
Comment (kylo252) [inline lua/lvim/core/nvimtree.lua:81]: This has repalced `quit_on_open = 0,`

File: lua/lvim/core/nvimtree.lua [hunk 3]
```diff
@@ -14,6 +14,12 @@ function M.config()
         "dashboard",
         "alpha",
       },
+      auto_reload_on_write = true,
+      hijack_unnamed_buffer_when_opening = false,
+      hijack_directories = {
```
Comment (kylo252) [inline lua/lvim/core/nvimtree.lua:19]: I can see that `project.nvim` has changed the requirements, and it doesn't seem to mention netrw anymore, https://github.com/ahmedkhalf/project.nvim/blob/cef52b8da07648b750d7f1e8fb93f12cb9482988/README.md. 

Can you test this please
```diff
diff --git a/lua/lvim/core/nvimtree.lua b/lua/lvim/core/nvimtree.lua
index 45dc950..54622f1 100644
--- a/lua/lvim/core/nvimtree.lua
+++ b/lua/lvim/core/nvimtree.lua
@@ -132,9 +132,10 @@ function M.setup()
   if lvim.builtin.project.active then
     lvim.builtin.nvimtree.respect_buf_cwd = 1
     lvim.builtin.nvimtree.setup.update_cwd = true
-    lvim.builtin.nvimtree.setup.disable_netrw = false
-    lvim.builtin.nvimtree.setup.hijack_netrw = false
-    vim.g.netrw_banner = false
+    lvim.builtin.nvimtree.setup.update_focused_file = {
+      enable = true,
+      update_cwd = true,
+    }
   end
```

File: lua/lvim/core/nvimtree.lua [hunk 4]
```diff
@@ -123,16 +137,22 @@ function M.setup()
     vim.g.netrw_banner = false
   end
 
+  local function telescope_find_files(_)
+    require("lvim.core.nvimtree").start_telescope "find_files"
+  end
+  local function telescope_find_str(_)
```
Comment (kylo252) [inline lua/lvim/core/nvimtree.lua:143]:
```suggestion
  local function telescope_live_grep(_)
```

File: lua/lvim/core/nvimtree.lua [hunk 5]
```diff
@@ -123,16 +137,22 @@ function M.setup()
...
-      { key = "v", cb = tree_cb "vsplit" },
-      { key = "C", cb = tree_cb "cd" },
-      { key = "gtf", cb = "<cmd>lua require'lvim.core.nvimtree'.start_telescope('find_files')<cr>" },
-      { key = "gtg", cb = "<cmd>lua require'lvim.core.nvimtree'.start_telescope('live_grep')<cr>" },
+      { key = { "l", "<CR>", "o" }, action = "edit", mode = "n" },
+      { key = "h", action = "close_node" },
+      { key = "v", action = "vsplit" },
+      { key = "C", action = "cd" },
+      { key = "gtf", action_cb = telescope_find_files },
+      { key = "gtg", action_cb = telescope_find_str },
```
Comment (kylo252) [inline lua/lvim/core/nvimtree.lua:155]:
```suggestion
      { key = "gtf", action = "telescope_find_files", action_cb = telescope_find_files },
      { key = "gtg", action = "telescope_live_grep", action_cb = telescope_live_grep },
```
Apparently the `action` tag is needed for some reason..

File: lua/lvim/core/nvimtree.lua [hunk 6]
```diff
@@ -75,7 +94,6 @@ function M.config()
       folder_arrows = 1,
       tree_width = 30,
     },
-    quit_on_open = 0,
     git_hl = 1,
     disable_window_picker = 0,
```
Comment (kylo252) [inline lua/lvim/core/nvimtree.lua:98]:
```suggestion
```

File: lua/lvim/core/nvimtree.lua [hunk 7]
```diff
@@ -67,6 +73,19 @@ function M.config()
...
+        change_dir = {
+          global = false,
+        },
+        open_file = {
+          quit_on_open = false,
+        },
+        window_picker = {
+          enable = false,
+          chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890",
+          exclude = {}, -- not sure if we need to define this or not
```
Comment (kylo252) [inline lua/lvim/core/nvimtree.lua:86]:
```suggestion
          exclude = {},
```
:sweat_smile:

File: lua/lvim/core/nvimtree.lua [hunk 8]
```diff
@@ -75,7 +94,6 @@ function M.config()
       folder_arrows = 1,
       tree_width = 30,
```
Comment (kylo252) [inline lua/lvim/core/nvimtree.lua:95]:
```suggestion
```

**Review by kylo252 — APPROVED** [review]
LGTM! :rocket:

**Comment by miguno** [comment]
FYI: I had the same problem described in https://github.com/LunarVim/LunarVim/issues/2302 again today, on a fresh (stable) install.

> Possibly Fixes https://github.com/LunarVim/LunarVim/issues/2302

Possibly not fixed. :)

**Comment by abzcoding** [comment]
> FYI: I had the same problem described in #2302 again today, on a fresh (stable) install.
> 
> > Possibly Fixes #2302
> 
> Possibly not fixed. :)

packer keeps force pushing and overriding their commit history, so we keep getting this error :(
but it might get better after we merge `rolling` into `master`

**Pull request merged**